packages/middleware-flexible-checksums/src/flexibleChecksumsInputMiddleware.ts

```diff
@@ -0,0 +1,57 @@
+import {
+  HandlerExecutionContext,
+  MetadataBearer,
+  RelativeMiddlewareOptions,
+  SerializeHandler,
+  SerializeHandlerArguments,
+  SerializeHandlerOutput,
+  SerializeMiddleware,
+} from "@smithy/types";
+
+import { PreviouslyResolved } from "./configuration";
+import { ResponseChecksumValidation } from "./constants";
+
+export interface FlexibleChecksumsInputMiddlewareConfig {
+  /**
+   * Defines a top-level operation input member used to opt-in to best-effort validation
+   * of a checksum returned in the HTTP response of the operation.
+   */
+  requestValidationModeMember?: string;
+}
+
+/**
+ * @internal
+ */
+export const flexibleChecksumsInputMiddlewareOptions: RelativeMiddlewareOptions = {
+  name: "flexibleChecksumsInputMiddleware",
+  toMiddleware: "serializerMiddleware",
+  relation: "before",
+  tags: ["BODY_CHECKSUM"],
+  override: true,
+};
+
+/**
+ * @internal
+ *
+ * The input counterpart to the flexibleChecksumsMiddleware.
+ */
+export const flexibleChecksumsInputMiddleware =
+  (
+    config: PreviouslyResolved,
+    middlewareConfig: FlexibleChecksumsInputMiddlewareConfig
+  ): SerializeMiddleware<any, any> =>
+  <Output extends MetadataBearer>(
+    next: SerializeHandler<any, Output>,
+    context: HandlerExecutionContext
+  ): SerializeHandler<any, Output> =>
+  async (args: SerializeHandlerArguments<any>): Promise<SerializeHandlerOutput<Output>> => {
+    const input = args.input;
+    const { requestValidationModeMember } = middlewareConfig;
+    const responseChecksumValidation = await config.responseChecksumValidation();
+
+    if (requestValidationModeMember && responseChecksumValidation === ResponseChecksumValidation.WHEN_SUPPORTED) {
+      input[requestValidationModeMember] = "ENABLED";
```

Comment:
what happens if the `if` conditions are true but `input[requestValidationModeMember]` is set by the user to some other value than `ENABLED` (ignoring the fact that this is the only enum value)? 

I would expect us to allow the value through and allow the user to receive a validation exception from the server.

Reply:
The `input[requestValidationModeMember]` is only allowed to be `ENABLED` as per specification.

This is what the requirements say
> Check if the user set requestValidationModeMember to ENABLED:
> * If they did, continue.
> * If they didn’t, check the value of ResponseChecksumValidation:
>   * If it's set to WHEN_SUPPORTED, set requestValidationModeMember to ENABLED and continue.
>   * If it's set to WHEN_REQUIRED, go to END.

To strictly satisfy the requirements, we should to check `input[requestValidationModeMember]` is not set to `ENABLED`. But it doesn't matter, since the value will be overridden to be `ENABLED` anyway.